platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java

```diff
@@ -127,10 +144,26 @@ public void setInitialState(@NonNull final MerkleRoot state) {
     public void handleConsensusRound(final ConsensusRound round) {
         final MerkleRoot state = stateRef.get();
 
-        uptimeTracker.handleRound(
-                round,
-                state.getPlatformState().getUptimeData(),
-                state.getPlatformState().getAddressBook());
+        if (!round.isEmpty()) {
```

Comment:
I don't think this should be in the `SwirldStateManager`. Instead, I think this actually should consume self events filtered from the consensus events coming out of the `ConsensusEngine`. Unfortunately, making this change would require a design proposal and thus probably won't be permitted at this time.

What do you think about this interim step instead?

- Bundle this functionality into a java class named something like `SelfEventMonitor`
    - There can also be some simplifications made to this code, since we don't actually need the most recent time any events came to consensus apart from self events. So the hash map isn't needed anymore
- Leave the call to `SelfEventMonitor` here in the `SwirldStateManager` for now
- Maybe put a `future work` comment, describing where this functionality ought to live

Reply:
I limited the scope of the change to removal `uptimeData` from the `PlatformState`. Now `UptimeTracker` has transient state of uptime. That is, it's in-memory and it gets lost with the node restart.